custom/src/main/java/co/elastic/otel/ElasticDistroResourceProvider.java

```diff
@@ -31,13 +31,21 @@ public class ElasticDistroResourceProvider implements ResourceProvider {
 
   @Override
   public Resource createResource(ConfigProperties configProperties) {
-    return AgentVersion.VERSION == null
-        ? Resource.empty()
-        : Resource.create(
-            Attributes.of(
-                ElasticAttributes.TELEMETRY_DISTRO_NAME,
-                "elastic",
-                ElasticAttributes.TELEMETRY_DISTRO_VERSION,
-                AgentVersion.VERSION));
+    if (AgentVersion.VERSION == null) {
```

Comment:
[question] should we provide a way to identify the extension version and name using maybe non-semconv attributes ? on the server/storage side it's often tricky to rely on "feature detection" based only on data shape or contents.